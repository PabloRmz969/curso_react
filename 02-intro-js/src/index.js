import heroes, {owners} from "./data/heroes";


const getHeroeById = (owner) =>  heroes.filter((heroe) => heroe.owner == owner);

console.log(getHeroeById('DC'));