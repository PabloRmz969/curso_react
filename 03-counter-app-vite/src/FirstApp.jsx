import PropTypes from 'prop-types'
const newMessage = {
	message: 'Hola Mundo',
	title: 'César'
}
const hiFunction = () => {
  	return "Hello world";
}
const multiplication = (a,b) => {
  	return a * b;
}
export const FirstApp = ( { title,subTitle,name } ) => {
	//Las funciones se agregan dentro de este scope si se necesita llamar después de un click o useState
	//No se pueden imprimir objetos
	return (
		<>
			<h1 data-testid='test-title'> { title } </h1>
			{/*<p>{ multiplication(5,4) }</p>*/}
			<p>{ subTitle }</p>
			<p>{ subTitle }</p>
			<p>{ name }</p>
			{/*<code>{JSON.stringify(newMessage)}</code>*/}
			<p>{name}</p>
		</>
    
	)
}

FirstApp.propTypes = {
	title: PropTypes.string.isRequired,
	subTitle: PropTypes.string.isRequired,
	name: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
	// title: 'No hay título',
	subTitle: 'No hay sub título',
	name: 'César Ramírez'
}