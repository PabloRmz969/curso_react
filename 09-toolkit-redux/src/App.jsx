import { useDispatch, useSelector } from 'react-redux'
import { decrement, increment } from './store/slices/counter';

import reactLogo from './assets/react.svg'
import viteLogo from '/vite.svg'
import './App.css'




function App() {
    const { counter } = useSelector(state => state.counter);
    const distpatch = useDispatch();

    return (
        <>
            <div>
                <a href="https://vitejs.dev" target="_blank">
                    <img src={viteLogo} className="logo" alt="Vite logo" />
                </a>
                <a href="https://react.dev" target="_blank">
                    <img src={reactLogo} className="logo react" alt="React logo" />
                </a>
            </div>
            <h1>Counter is {counter}</h1>
            <div className="card">
                <button onClick={() => distpatch( increment() )}>
                    Increment
                </button>
                <button onClick={() => distpatch( decrement() )}>
                    Decrement
                </button>
                <button onClick={() => distpatch( increment(2) )}>
                    Increment by 2
                </button>
            </div>
        </>
    )
}

export default App
