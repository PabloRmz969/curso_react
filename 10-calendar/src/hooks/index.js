export * from './useModal';
export * from './useUiStore';
export * from './useCalendarStore';