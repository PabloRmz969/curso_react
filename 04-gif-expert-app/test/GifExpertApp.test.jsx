import { fireEvent, render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe('pruebas en <GifExpertApp/>', () => {
    test('debe hacer match con el snapshot', () => {
        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
    })

    // test('probando onAddCategory', () => {
    //     const inputValue = 'Saitama';
    //     const onAddCategory = jest.fn('Naruto');

    //     render(<GifExpertApp />);
    //     expect(onAddCategory).toHaveBeenCalled();
        
    // })
})