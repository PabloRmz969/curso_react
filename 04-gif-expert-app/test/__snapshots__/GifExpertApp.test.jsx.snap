// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifExpertApp/> debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Jujutsu kaien
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
  <h3>
    Boku no hero
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
